### PR TITLE
Inject W3C headers in HTTP diagnostics handler

### PR DESCRIPTION
**Pull request by lmolkova**
Depending on the Activity.IdFormat we want to inject [W3C distributed tracing](https://github.com/w3c/trace-context) headers or Request-Id headers.

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs [hunk 1]
```diff
@@ -380,6 +380,83 @@ public void SendAsync_ExpectedDiagnosticSourceActivityLogging()
...
+
+                Activity parentActivity = new Activity("parent");
+                parentActivity.SetParentId(ActivityTraceId.CreateRandom(), ActivitySpanId.CreateRandom());
+                parentActivity.AddBaggage("moreBaggage", Guid.NewGuid().ToString());
+                parentActivity.Start();
+
+                var diagnosticListenerObserver = new FakeDiagnosticListenerObserver(kvp =>
+                {
+                    if (kvp.Key.Equals("System.Net.Http.Request"))
+                    { requestLogged = true; }
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs]: This is not standard bracing style for CoreFx.

File: src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs [hunk 2]
```diff
@@ -78,9 +78,21 @@ internal static bool IsEnabled()
...
-                request.Headers.Add(DiagnosticsHandlerLoggingStrings.RequestIdHeaderName, currentActivity.Id);
+                if (currentActivity.IdFormat == ActivityIdFormat.W3C && !request.Headers.Contains(DiagnosticsHandlerLoggingStrings.TraceParentHeaderName))
+                {
+                    request.Headers.Add(DiagnosticsHandlerLoggingStrings.TraceParentHeaderName, currentActivity.Id);
+                    if (currentActivity.TraceStateString != null)
+                    {
+                        request.Headers.Add(DiagnosticsHandlerLoggingStrings.TraceStateHeaderName, currentActivity.TraceStateString);
+                    }
+                }
+                else if(!request.Headers.Contains(DiagnosticsHandlerLoggingStrings.RequestIdHeaderName))
```
Comment (vancem) [inline src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs]: Do you want the 'if' condition to have two terms in it?   
As it is if the ID format is W3C AND you already have a TraceParent header, you ALSO try to add a RequestId header.   

This is not what I would have expected.   Feels like what you want is 
```
                if (currentActivity.IdFormat == ActivityIdFormat.W3C)
                {
                    if (request.Headers.Contains(DiagnosticsHandlerLoggingStrings.TraceParentHeaderName))
                    {
                        request.Headers.Add(DiagnosticsHandlerLoggingStrings.TraceParentHeaderName, currentActivity.Id);
                        if (currentActivity.TraceStateString != null)
                        {
                            request.Headers.Add(DiagnosticsHandlerLoggingStrings.TraceStateHeaderName, currentActivity.TraceStateString);
                        }
                    }
                }
                else // Not W3C Format
                { 
                    if(!request.Headers.Contains(DiagnosticsHandlerLoggingStrings.RequestIdHeaderName))
                    {
                        request.Headers.Add(DiagnosticsHandlerLoggingStrings.RequestIdHeaderName, currentActivity.Id);
                    }
                }
```

Comment (vancem) [reply src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs]: But otherwise it looks OK.

Comment (lmolkova) [reply src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs]: thanks, fixed

**Comment by vancem** [comment]
+ @noahfalk 

This change, if you ignore the testing changes, ia quite small and simply propagating the new W3C ids into HTTP headers.  It is doing the analogous thing to what was done before or previous (hierarachical) IDs.  This is new behavior that should be back-compat.   

Unless someone raises an objection I will merge this later today.

**Pull request merged**